hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -230,21 +231,19 @@ chainSyncHandler tracer callback getTimeHandle ctx =
                       }
                 }
 
--- | Hardcoded grace time for close transaction to be valid.
--- TODO: make it a node configuration parameter
-closeGraceTime :: SlotNo
-closeGraceTime = 100
-
-fromPostChainTx ::
+prepareTxToPost ::
   (MonadSTM m, MonadThrow (STM m)) =>
   TimeHandle ->
   TinyWallet m ->
   ChainContext ->
   ChainStateType Tx ->
   PostChainTx Tx ->
   STM m Tx
-fromPostChainTx timeHandle wallet ctx cst@ChainStateAt{chainState} tx = do
-  pointInTime <- throwLeft currentPointInTime
+prepareTxToPost timeHandle wallet ctx cst@ChainStateAt{chainState} tx = do
+  pointInTime@(currentSlot, _) <- throwLeft currentPointInTime
+  let (UnsafeContestationPeriod cpNatural) = contestationPeriod ctx
+      -- we are using the contestationPeriod as tx close grace time. See ADR21 for context
+      closeGraceTime = SlotNo $ fromIntegral $ cpNatural * 2
```

Comment:
This should not be 2*cp! The validator should fail on anything `> cp`

Reply:
Yes, with the new changes this is no longer valid/needed.

Reply:
BUT we need to bump contestation period in end-to-end tests from value 1 to 2 for example since I observed failed tests where upper tx bound ends up being the same slot in which the validation happens.